src/NuGet.Clients/NuGet.CommandLine/Commands/InstallCommand.cs

```diff
@@ -209,6 +212,17 @@ private CommandLineSourceRepositoryProvider GetSourceRepositoryProvider()
             return new CommandLineSourceRepositoryProvider(SourceProvider);
         }
 
+        private DependencyBehavior GetSettingsDependencyBehavior(Configuration.ISettings settings)
+        {
+            DependencyBehavior ret;
+
+            string depSetting = SettingsUtility.GetConfigValue(settings, "dependencyVersion");
+            if (!Enum.TryParse<DependencyBehavior>(depSetting, out ret))
+                ret = DependencyBehavior.Lowest; // if fails, set to default behavior
```

Comment:
It seems like this one should fail also, you can create a common method that handles parsing the flag or failing with the new exception message.